cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/SecureStoreHandler.java

```diff
@@ -0,0 +1,113 @@
+import co.cask.cdap.proto.security.SecureStoreCreateRequest;
+import co.cask.http.HttpResponder;
+import com.google.inject.Inject;
+import org.jboss.netty.handler.codec.http.HttpRequest;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+
+import java.io.IOException;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+/**
+ * Exposes REST APIs for {@link co.cask.cdap.api.security.store.SecureStore} and
+ * {@link co.cask.cdap.api.security.store.SecureStoreManager}.
+ */
+@Path(Constants.Gateway.API_VERSION_3 + "/security/store/namespaces/{namespace-id}")
+public class SecureStoreHandler extends AbstractAppFabricHttpHandler {
+
+  private final SecureStore secureStore;
+  private final SecureStoreManager secureStoreManager;
+
+  @Inject
+  public SecureStoreHandler(SecureStore secureStore, SecureStoreManager secureStoreManager) {
+    this.secureStore = secureStore;
+    this.secureStoreManager = secureStoreManager;
+  }
+
+  @Path("/key")
+  @PUT
+  public void create(HttpRequest httpRequest, HttpResponder httpResponder) throws BadRequestException {
+    // Check authentication
+    SecureStoreCreateRequest secureStoreCreateRequest = parseBody(httpRequest, SecureStoreCreateRequest.class);
+
+    if (secureStoreCreateRequest == null) {
+      throw new BadRequestException("Unable to parse the request.");
+    }
+
+    String name = secureStoreCreateRequest.getName();
+    String description = secureStoreCreateRequest.getDescription();
+    byte[] data = secureStoreCreateRequest.getData();
+    if (name == null || data == null) {
+      throw new BadRequestException("Name or data empty.");
+    }
+    try {
+      secureStoreManager.put(name, data, description, secureStoreCreateRequest.getProperties());
+    } catch (IOException e) {
+      httpResponder.sendString(HttpResponseStatus.INTERNAL_SERVER_ERROR, "Failed to store the item in the store.");
+      return;
+    }
+    httpResponder.sendStatus(HttpResponseStatus.OK);
+  }
+
+  @Path("/keys/{key-name}")
+  @DELETE
+  public void delete(HttpRequest httpRequest, HttpResponder httpResponder, @PathParam("key-name") String name) {
+    try {
+      secureStoreManager.delete(name);
+    } catch (IOException e) {
+      httpResponder.sendString(HttpResponseStatus.INTERNAL_SERVER_ERROR, "Failed to delete the key.");
+      return;
+    }
+    httpResponder.sendStatus(HttpResponseStatus.OK);
+  }
+
+  @Path("/keys/{key-name}")
+  @GET
+  public void get(HttpRequest httpRequest, HttpResponder httpResponder, @PathParam("key-name") String name) {
+    SecureStoreData secureStoreData;
+    try {
+      secureStoreData = secureStore.get(name);
+    } catch (IOException e) {
+      httpResponder.sendString(HttpResponseStatus.INTERNAL_SERVER_ERROR, "Failed to get the key.");
+      return;
+    }
+    httpResponder.sendJson(HttpResponseStatus.OK, secureStoreData.get());
```

Comment:
Doesn't seem like it. its needed.
![image](https://cloud.githubusercontent.com/assets/3895099/16674644/040e66be-446f-11e6-8ac2-e2f9be8fd21d.png)